### ISSUE TITLE
Revive no-op insertion pass for RISC-V

### ISSUE THREAD

**Issue by ArsenyBochkarev** [Compiler-assisted-fuzzing/llvm-project#4]
You'll need to revive the reverted [random `nop`s insertion pass](https://reviews.llvm.org/D3392). 
High-level steps
- Read the review
- Reproduce the patch
- Transfer the patch to RISC-V
- Verify the pass actually inserts `nop`s by using [this](https://stackoverflow.com/questions/11227809/why-is-processing-a-sorted-array-faster-than-processing-an-unsorted-array?m=1) test (`objdump -d` for RISC-V target).

**Comment by d-tolmachev** [comment]
Consider the following code snippet:
```
    const unsigned arraySize = 32768;
    int data[arraySize];
    for (unsigned c = 0; c < arraySize; ++c) {
        data[c] = std::rand() % 256;
    }
    long long sum = 0;
    for (unsigned i = 0; i < 100000; ++i) {
        for (unsigned c = 0; c < arraySize; ++c) {
            if (data[c] >= 128) {
                sum += data[c];
            }
        }
    }
```
Due to the randomness of the values ​​of array elements, a significant proportion of BranchPredictor assumptions are unsuccessful. The effect of this pass is to add nop-instructions to random places in the machine code.
The pass can be triggered when compiled with the `-fdiversify -O0` flags.
Diff of the code obtained after applying this pass, with the version compiled in normal mode, is in the attachments:
[NoopInsertionPass_test_withnops_withoutnops_diff.txt](https://github.com/user-attachments/files/17821465/NoopInsertionPass_test_withnops_withoutnops_diff.txt)

**Comment by ArsenyBochkarev** [comment]
Closed by #11
